packages/template-block/src/TemplateBlock.tsx

```diff
@@ -195,9 +195,12 @@ export const TemplateBlock = ({ appBridge }: BlockProps): ReactElement => {
                                 <TemplateText
                                     appBridge={appBridge}
                                     title={title}
-                                    blockSettings={blockSettings}
                                     description={description}
-                                    pageCount={selectedTemplate?.pages.length ?? 0}
+                                    pageCount={
+                                        blockSettings.hasPageCount !== false
+                                            ? selectedTemplate?.pages.length ?? 0
+                                            : undefined
```

Comment:
```suggestion
                                        blockSettings.hasPageCount
                                            ? selectedTemplate?.pages.length ?? 0
                                            : undefined
```

Minor but hasPageCount should have it's default value populated when the block is created, so it doesn't need to consider `true | undefined` because it should only be able to be boolean.

Reply:
Thanks for the hint! 🚀